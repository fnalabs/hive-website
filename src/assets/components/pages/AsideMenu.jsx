import React from 'react'
import { NavLink } from 'react-router-dom'

import { AnalyticsLink } from 'common'

const AsideMenu = () => (
  <aside className='menu'>
    <p className='menu-label has-text-weight-bold'>v2.1.3</p>
    <ul className='menu-list'>
      <li><NavLink to='/overview' activeClassName='is-active'>Overview</NavLink>
        <ul>
          <li><NavLink to='/model' activeClassName='is-active'>Data Model</NavLink></li>
          <li><NavLink to='/domain' activeClassName='is-active'>Domain Logic</NavLink></li>
          <li><NavLink to='/infrastructure' activeClassName='is-active'>Infrastructure</NavLink></li>
          <li><NavLink to='/telemetry' activeClassName='is-active'>Telemetry</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/start' activeClassName='is-active'>Get Started</NavLink>
        <ul>
          <li><NavLink to='/setup' activeClassName='is-active'>Setup</NavLink></li>
          <li className='menu-label has-text-weight-bold'>Examples:</li>
          <li><NavLink to='/basic' activeClassName='is-active'>Basic</NavLink></li>
          <li><NavLink to='/rest' activeClassName='is-active'>Rest</NavLink></li>
          <li><NavLink to='/cqrs-es' activeClassName='is-active'>CQRS/ES</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/documentation' activeClassName='is-active'>Documentation</NavLink>
        <ul>
          <li><NavLink to='/environments' activeClassName='is-active'>Environments</NavLink></li>
          <li><AnalyticsLink to='https://fnalabs.github.io/hive-io/' target='_blank' rel='noopener noreferrer'>API</AnalyticsLink></li>
        </ul>
      </li>
    </ul>
  </aside>
)

export default AsideMenu
