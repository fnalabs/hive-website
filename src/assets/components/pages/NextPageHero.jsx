import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { AnalyticsLink } from 'common'
import { Hero, HeroBody } from 'common/hero'
import { ArrowRight } from 'icons'

const toRegex = /^https:\/\//

export default class NextPageHero extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired
  }

  render () {
    const to = this.props.to

    return (
      <Hero size='small' className='is-hive'>
        <HeroBody centered>
          {toRegex.test(to)
            ? <AnalyticsLink to={to}><ArrowRight height='48' /></AnalyticsLink>
            : <Link to={to}><ArrowRight height='48' /></Link>}
        </HeroBody>
      </Hero>
    )
  }
}
