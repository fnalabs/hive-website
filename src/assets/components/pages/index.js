// site's main pages
export { default as Home } from './home/Home.jsx'
export { default as Overview } from './overview/Overview.jsx'
export { default as Start } from './start/Start.jsx'

// policy pages
export { Cookie } from './policies/Cookie.jsx'
export { Privacy } from './policies/Privacy.jsx'
