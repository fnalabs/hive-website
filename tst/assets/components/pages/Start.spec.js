/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { MemoryRouter } from 'react-router'
import ReactGA from 'react-ga'

import { Start } from 'pages'

jest.mock('react-ga')

describe('<Start />', () => {
  it('should render Start page static content', () => {
    const tree = renderer.create(<MemoryRouter><Start location={{ pathname: '/start' }} /></MemoryRouter>).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).not.toBeCalled()
  })

  it('should render Start page static content and report page view', () => {
    shallow(<Start location={{ pathname: '/start' }} />, {
      context: { isConsent: true }
    })

    expect(ReactGA.pageview).toBeCalledTimes(1)
  })
})
