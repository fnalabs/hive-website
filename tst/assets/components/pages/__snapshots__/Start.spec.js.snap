// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Start /> should render Start page static content 1`] = `
Array [
  <article
    className="section is-medium"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <header
          className="column is-narrow is-hidden-touch is-aside"
        >
          <aside
            className="menu"
          >
            <p
              className="menu-label has-text-weight-bold"
            >
              v2.0.0-rc.3
            </p>
            <ul
              className="menu-list"
            >
              <li>
                <a
                  aria-current={null}
                  href="/overview"
                  onClick={[Function]}
                >
                  Overview
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/model"
                      onClick={[Function]}
                    >
                      Data Model
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/domain"
                      onClick={[Function]}
                    >
                      Domain Logic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/infrastructure"
                      onClick={[Function]}
                    >
                      Infrastructure
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/start"
                  onClick={[Function]}
                >
                  Get Started
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/setup"
                      onClick={[Function]}
                    >
                      Setup
                    </a>
                  </li>
                  <li
                    className="menu-label has-text-weight-bold"
                  >
                    Examples:
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/basic"
                      onClick={[Function]}
                    >
                      Basic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/rest"
                      onClick={[Function]}
                    >
                      Rest
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/cqrs-es"
                      onClick={[Function]}
                    >
                      CQRS/ES
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/documentation"
                  onClick={[Function]}
                >
                  Documentation
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/environments"
                      onClick={[Function]}
                    >
                      Environments
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-js/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      API
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </aside>
        </header>
        <section
          className="column content"
          role="document"
        >
          <h1>
            Get Started
          </h1>
          <p>
            The next page contains prerequisites and installation instructions to get you started with Hive
            <sup>
              io
            </sup>
            . The examples provided cover the basics of RESTful and CQRS/ES architectures. Descriptions are also provided on how to extend the concepts to future iterations to express the evolution of a project.
          </p>
          <div
            className="notification"
          >
            <span
              className="icon"
            >
              <svg
                className="svg-inline"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                  fill="#209cee"
                />
              </svg>
            </span>
            <span>
              The Development process for domain logic only requires a Text Editor or IDE that supports JavaScript and Node.js/Docker. The rest of the instructions outlined here provide infrastructure details for installation and deployment.
            </span>
          </div>
          <h2>
            Prerequisites
          </h2>
          <p>
            The prerequisites vary depending on the architecture. All implementations require Docker of some form. Whether it is just pure Docker, Docker Swarm, or Kubernetes is up to you. It is also strongly recommended you use a load balancer of some sort since the whole purpose of the framework is horizontal scalability.
          </p>
          <h3>
            Cloud Platforms
          </h3>
          <dl>
            <dt>
              <em>
                <strong>
                  Required
                </strong>
              </em>
            </dt>
            <dd>
              <a
                href="https://www.docker.com/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Docker
              </a>
            </dd>
            <dt>
              <em>
                <strong>
                  Recommended
                </strong>
              </em>
            </dt>
            <dd>
              <a
                href="https://kubernetes.io/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Kubernetes
              </a>
              , 
              <a
                href="https://docs.docker.com/engine/swarm/"
                rel="noopener noreferrer"
                target="_blank"
              >
                Swarm
              </a>
              , or a 
              <a
                href="https://en.wikipedia.org/wiki/Category:Cloud_computing_providers"
                rel="noopener noreferrer"
                target="_blank"
              >
                cloud service provider
              </a>
            </dd>
          </dl>
          <h3>
            More
          </h3>
          <p>
            Once you decide on your cloud platform, you're ready to start building and deploying your application services and dependencies. Hive
            <sup>
              io
            </sup>
             will integrate well with many of the 
            <a
              href="https://www.cncf.io/projects/"
              rel="noopener noreferrer"
              target="_blank"
            >
              cloud-native projects
            </a>
             and 
            <a
              href="https://hub.docker.com/"
              rel="noopener noreferrer"
              target="_blank"
            >
              other containerized services
            </a>
            .
          </p>
        </section>
      </div>
    </div>
  </article>,
  <section
    className="hero is-small is-hive"
  >
    <div
      className="hero-body"
    >
      <nav
        className="level is-mobile"
      >
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="previous page"
            href="/infrastructure"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                />
              </svg>
            </span>
          </a>
        </div>
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="next page"
            href="/setup"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                />
              </svg>
            </span>
          </a>
        </div>
      </nav>
    </div>
  </section>,
]
`;
